### PR TITLE
Revamp Filters, Support for HTTP Filters, AccessLog Filters

### PR DESCRIPTION
**Pull request by ajaypj**

### CODE REVIEW

File: core/src/main/java/com/flipkart/gjex/core/filter/GrpcAccessLogFilter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    String size = null;
+    if (response != null){
+      size = String.valueOf(response.getSerializedSize());
+    }
+    StringBuilder sb = new StringBuilder()
+        .append(serverRequestParams.getClientIp()).append(" ")
+        .append(serverRequestParams.getMethodName()).append(" ")
+        .append(size).append(" ")
+        .append(System.currentTimeMillis()-startTime);
+    error(sb.toString());
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/GrpcAccessLogFilter.java:45]: why error? here

File: core/src/main/java/com/flipkart/gjex/core/web/filter/HttpAccessLogFilter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+@Singleton
+@Named("HttpAccessLogFilter")
+public class HttpAccessLogFilter implements Filter, Logging {
+  private long startTime;
+  @Override
+  public void init(FilterConfig filterConfig) throws ServletException {}
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/web/filter/HttpAccessLogFilter.java:31]: We should change the signature to match how the grpc filter has

`public void doProcessRequest(R request) `

and 

`public void doProcessResponse(S response)`

File: core/src/main/java/com/flipkart/gjex/core/filter/GrpcAccessLogFilter.java (hunk 1 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/GrpcAccessLogFilter.java:45]: added access-log separately  and made it info

File: examples/build.gradle [hunk 3]
```diff
@@ -88,7 +89,7 @@ task runHelloWorldServer(type: JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
     mainClass = 'com.flipkart.gjex.examples.helloworld.HelloWorldApplication'
     args "server", "./src/main/resources/hello_world_config.yml"
-    jvmArgs '--add-opens=java.base/java.lang=ALL-UNNAMED'
+//     jvmArgs '--add-opens=java.base/java.lang=ALL-UNNAMED'
```
Comment (kingster) [inline examples/build.gradle:92]: uncheck this line.

File: guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java [hunk 4]
```diff
@@ -82,8 +83,12 @@ public void registerFilters(List<Filter> filters, List<BindableService> services
...
                     // Key is of the form <Service Name>+ "/" +<Method Name>
                     // reflecting the structure followed in the gRPC HandlerRegistry using MethodDescriptor#getFullMethodName()
-                    filtersMap.put((service.bindService().getServiceDescriptor().getName() + "/" + pair.getValue().getName()).toLowerCase(),
-                            filtersForMethod);
+                    String methodSignature =
+                        (service.bindService().getServiceDescriptor().getName() + "/" + pair.getValue().getName()).toLowerCase();
+                    filtersMap.put(methodSignature, filtersForMethod);
+                    if (enableAccessLogs){
+                        filtersMap.get(methodSignature).add(new GrpcAccessLogFilter());
+                    }
```
Comment (kingster) [inline guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java:91]: why not do this before, and get it added in `filtersForMethod` as required

File: guice/src/main/java/com/flipkart/gjex/grpc/service/ApiServer.java [hunk 5]
```diff
@@ -40,19 +44,39 @@
...
 	public void registerResources(List<ResourceConfig> resourceConfigs) {
 		this.resourceConfigs.addAll(resourceConfigs);
 	}
+
+	public void registerHttpFilters(List<HttpFilterParams> httpFilterParamsList,
+																	boolean enableAccessLogs){
+		if (enableAccessLogs) {
+			context.addFilter(new FilterHolder(accessLogFilter), "/*" ,
+					EnumSet.of(DispatcherType.REQUEST));
+		}
```
Comment (kingster) [inline guice/src/main/java/com/flipkart/gjex/grpc/service/ApiServer.java:73]: we should make extract this out to a private separate function

File: guice/src/main/java/com/flipkart/gjex/guice/GuiceBundle.java [hunk 6]
```diff
@@ -150,7 +150,7 @@ public void run(T configuration, U configMap, Environment environment) {
 
 		// Add all Grpc Filters to the Grpc Server
 		filters = getInstances(baseInjector, Filter.class);
-		grpcServer.registerFilters(filters, bindableServices);
+		grpcServer.registerFilters(filters, bindableServices, configuration.getGrpc().isEnableAccessLogs());
```
Comment (kingster) [inline guice/src/main/java/com/flipkart/gjex/guice/GuiceBundle.java:153]: lets not pass boolean in arguments. Arguments should be proper class or objects.

File: core/src/main/java/com/flipkart/gjex/core/web/filter/HttpAccessLogFilter.java (hunk 2 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/web/filter/HttpAccessLogFilter.java]: done

File: guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java (hunk 4 above)
Comment (ajaypj) [reply guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java]: done

File: guice/src/main/java/com/flipkart/gjex/grpc/service/ApiServer.java (hunk 5 above)
Comment (ajaypj) [reply guice/src/main/java/com/flipkart/gjex/grpc/service/ApiServer.java]: done

File: guice/src/main/java/com/flipkart/gjex/guice/GuiceBundle.java (hunk 6 above)
Comment (ajaypj) [reply guice/src/main/java/com/flipkart/gjex/guice/GuiceBundle.java]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/GjexFilter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.flipkart.gjex.core.logging.Logging;
+
+/**
+ * A Filter interface for processing Request, Request-Headers, Response and Response-Headers 
+ * around gRPC and HTTP method invocation
+ *
+ * @author ajay.jalgaonkar
+ */
+public abstract class GjexFilter<Req, Res, M> implements Logging {
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/GjexFilter.java:26]: Remove the prefix `Gjex`. Classes have aboslute names, so we don't need to prefix `Gjex` since the class path itself has `gjex`

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/GjexGrpcFilter.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * A Filter interface for processing Request, Request-Headers, Response and Response-Headers around gRPC method invocation
+ *  
+ * @author ajay.jalgaonkar
+ *
+ * @param <Req> Proto V3 message
+ * @param <Res> Proto V3 message
+ */
+public abstract class GjexGrpcFilter<Req extends GeneratedMessageV3, Res extends GeneratedMessageV3>
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/grpc/GjexGrpcFilter.java:30]: Remove the prefix `Gjex`. Classes have aboslute names, so we don't need to prefix `Gjex` since the class path itself has `gjex`

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/GrpcAccessLogGjexGrpcFilter.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import com.flipkart.gjex.core.logging.Logging;
+import com.google.protobuf.GeneratedMessageV3;
+import io.grpc.Metadata;
+
+/**
+ * Filter for logging grpc access log requests
+ * @author ajay.jalgaonkar
+ *
+ */
+public class GrpcAccessLogGjexGrpcFilter<R extends GeneratedMessageV3,
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/grpc/GrpcAccessLogGjexGrpcFilter.java:14]: Too much-repeated names?`GrpcAccessLogFilter` should have been enough.

File: core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * An Http Filter extending {@link GjexFilter}  interface for processing Request, Request-Headers,
+ * Response and
+ * Response-Headers
+ * around HTTP method invocation
+ *
+ * @author ajay.jalgaonkar
+ */
+@Data
+public abstract class GjexHttpFilter extends GjexFilter<ServletRequest, ServletResponse,
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java:30]: please rename to `HttpFilter`

File: core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+  @Override
+  public void doProcessResponseHeaders(Set<String> responseHeaders) {
+    super.doProcessResponseHeaders(responseHeaders);
+  }
+
+  @Override
+  public void doProcessResponse(ResponseParams<ServletResponse> responseParams) {
+    super.doProcessResponse(responseParams);
+  }
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java:76]: why are you defining this methods? These needs to be implemented by the implementors of this abstract class.

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGjexGrpcFilter.java [hunk 12]
```diff
@@ -15,27 +15,26 @@
...
- * An implementation of the {@link Filter} interface as example that performs naive authentication based on
+ * An implementation of the {@link GjexGrpcFilter} interface as example that performs naive authentication based on
  * information contained in the Request headers
  *
  * @author regu.b
  *
  */
 @Named("AuthFilter")
-public class AuthFilter implements Filter<HelloRequest, HelloReply>, Logging {
+public class AuthGjexGrpcFilter extends GjexGrpcFilter<HelloRequest, HelloReply> implements Logging {
```
Comment (kingster) [inline examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGjexGrpcFilter.java:37]: Name correction please.

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGjexGrpcFilter.java [hunk 13]
```diff
@@ -15,31 +15,34 @@
...
 import javax.inject.Named;
 
 /**
- * An implementation of the {@link Filter} interface as example that simply logs Request information
+ * An implementation of the {@link GjexGrpcFilter} interface as example that simply logs Request information
  * @author regu.b
  */
 @Named("LoggingFilter")
-public class LoggingFilter<Req extends GeneratedMessageV3, Res extends GeneratedMessageV3> implements Filter<Req, Res>, Logging {
+public class LoggingGjexGrpcFilter<Req extends GeneratedMessageV3,
```
Comment (kingster) [inline examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGjexGrpcFilter.java:32]: Name correction please.

File: examples/src/main/resources/hello_world_config.yml [hunk 14]
```diff
@@ -1,6 +1,8 @@
 Grpc:
     server.port: 50051
     server.executorThreads : 4
+    grpcFilterConfig:
```
Comment (kingster) [inline examples/src/main/resources/hello_world_config.yml:4]: Access log can be a top level config.

File: examples/src/main/resources/hello_world_config.yml [hunk 15]
```diff
@@ -14,6 +16,8 @@ Api:
     service.selectors: 10
     service.workers: 30
     scheduledexecutor.threadpool.size: 1
+    httpFilterConfig:
```
Comment (kingster) [inline examples/src/main/resources/hello_world_config.yml:19]: Access log can be a top level config.

File: grpc-jexpress-template/src/main/java/com/flipkart/grpc/jexpress/SampleApplication.java [hunk 16]
```diff
@@ -1,36 +0,0 @@
...
-
-import com.flipkart.gjex.core.Application;
-import com.flipkart.gjex.core.setup.Bootstrap;
-import com.flipkart.gjex.core.setup.Environment;
-import com.flipkart.gjex.guice.GuiceBundle;
-import com.flipkart.grpc.jexpress.module.SampleModule;
-
-import java.util.Map;
-
-public class SampleApplication extends Application<SampleConfiguration, Map> {
```
Comment (kingster) [inline grpc-jexpress-template/src/main/java/com/flipkart/grpc/jexpress/SampleApplication.java:11]: Why have we deleted these files?

File: guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java [hunk 17]
```diff
@@ -82,58 +84,36 @@ public void registerFilters(List<Filter> filters, List<BindableService> services
...
-        listener = next.startCall(new SimpleForwardingServerCall<ReqT, RespT>(call) {
+        ServerCall.Listener<Req> listener = null;
+        listener = next.startCall(new SimpleForwardingServerCall<Req, Res>(call) {
             @Override
-            public void sendMessage(final RespT response) {
+            public void sendMessage(final Res response) {
                 Context previous = attachContext(contextWithHeaders);   // attaching headers to gRPC context
                 try {
-                    filters.forEach(filter -> filter.doProcessResponse((GeneratedMessageV3) response));
+                    grpcFilters.forEach(filter -> filter.doProcessResponse(ResponseParams.builder().response(response).build()));
```
Comment (kingster) [inline guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java:116]: why wrap? what is the usecase?

File: guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java [hunk 18]
```diff
@@ -166,10 +146,27 @@ public void onHalfClose() {
...
-                    filters.forEach(filter -> filter.doProcessRequest((GeneratedMessageV3) request));
+                    for (GjexGrpcFilter grpcFilter : grpcFilters) {
+                        if (grpcFilter != null) {
+                            RequestParams requestParams =
+                                RequestParams.builder()
+                                    .clientIp(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR).toString())
+                                    .resourcePath(call.getMethodDescriptor().getFullMethodName().toLowerCase())
+                                    .metadata(headers)
+                                    .request(request)
+                                    .build();
```
Comment (kingster) [inline guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java:160]: Why create this in loop? Can it not be outside?

**Review by kingster — CHANGES_REQUESTED** [review]

File: guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java (hunk 17 above)
Comment (ajaypj) [reply guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java:116]: ResponseParams is a common class used for both http and grpc responses. so wrapped the grpc "response"

File: examples/src/main/resources/hello_world_config.yml (hunk 14 above)
Comment (ajaypj) [reply examples/src/main/resources/hello_world_config.yml:4]: If there are other config params other than accesslog, then this filterconfig can be used.
did not want to keep accesslog related config separate

File: grpc-jexpress-template/src/main/java/com/flipkart/grpc/jexpress/SampleApplication.java (hunk 16 above)
Comment (ajaypj) [reply grpc-jexpress-template/src/main/java/com/flipkart/grpc/jexpress/SampleApplication.java]: added back

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGjexGrpcFilter.java (hunk 12 above)
Comment (ajaypj) [reply examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGjexGrpcFilter.java]: done

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGjexGrpcFilter.java (hunk 13 above)
Comment (ajaypj) [reply examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGjexGrpcFilter.java]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java (hunk 11 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java]: removed

File: core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java (hunk 10 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/http/GjexHttpFilter.java]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/GrpcAccessLogGjexGrpcFilter.java (hunk 9 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/grpc/GrpcAccessLogGjexGrpcFilter.java]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/GjexGrpcFilter.java (hunk 8 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/grpc/GjexGrpcFilter.java]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/GjexFilter.java (hunk 7 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/GjexFilter.java]: done

File: core/src/main/java/com/flipkart/gjex/core/config/ApiService.java [hunk 19]
```diff
@@ -38,43 +41,6 @@ public class ApiService {
...
-    }
-
-    public int getScheduledExecutorThreadPoolSize() {
-        return scheduledExecutorThreadPoolSize;
-    }
-
-    public void setScheduledExecutorThreadPoolSize(int scheduledExecutorThreadPoolSize) {
-        this.scheduledExecutorThreadPoolSize = scheduledExecutorThreadPoolSize;
-    }
+    @JsonProperty("httpFilterConfig")
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/config/ApiService.java:44]: should be named `filterConfig` since API Section itself is for HTTP

File: core/src/main/java/com/flipkart/gjex/core/config/GrpcConfig.java [hunk 20]
```diff
@@ -28,27 +31,6 @@ public class GrpcConfig {
...
-
-    public int getExecutorThreads() {
-        return executorThreads;
-    }
-
-    public void setExecutorThreads(int executorThreads) {
-        this.executorThreads = executorThreads;
-    }
+    @JsonProperty("grpcFilterConfig")
+    private GrpcFilterConfig grpcFilterConfig;
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/config/GrpcConfig.java:35]: Same here, should be `filterConfig` simply

File: core/src/main/java/com/flipkart/gjex/core/filter/Filter.java [hunk 21]
```diff
@@ -15,64 +15,36 @@
...
 /**
- * A Filter interface for processing Request, Request-Headers, Response and Response-Headers around gRPC method invocation
- *  
- * @author regu.b
+ * A Filter interface for processing Request, Request-Headers, Response and Response-Headers 
+ * around gRPC and HTTP method invocation
  *
- * @param <Req> Proto V3 message
- * @param <Res> Proto V3 message
+ * @author ajay.jalgaonkar
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/Filter.java:24]: Please dont change the original author information. Since there has been no new code here.

File: core/src/main/java/com/flipkart/gjex/core/filter/ResponseParams.java [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * Wrapper for response of Http/Grpc Request
+ *
+ * @author ajay.jalgaonkar
+ */
+
+@Getter
+@Builder
+public class ResponseParams <S>{
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/ResponseParams.java:14]: Why is this required?

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+    this.requestParams = requestParams;
+  }
+
+  public StringBuilder getSb() {
+    return sb;
+  }
+
+  public void setSb(StringBuilder sb) {
+    this.sb = sb;
+  }
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java:74]: Why are these required, given that `@Data` has been already used?

File: core/src/main/java/com/flipkart/gjex/core/config/ApiService.java (hunk 19 above)
Comment (kingster) [reply core/src/main/java/com/flipkart/gjex/core/config/ApiService.java:44]: Also, given that it contains a single boolean, may be we flatten it, and make accessLog as top level attribute.

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+  public void doProcessResponse(ResponseParams<S> responseParams) {
+    String size = null;
+    if (responseParams.getResponse() != null){
+      size = String.valueOf(responseParams.getResponse().getSerializedSize());
+    }
+    sb.append(requestParams.getClientIp()).append(" ")
+      .append(requestParams.getResourcePath()).append(" ")
+      .append(size).append(" ")
+      .append(System.currentTimeMillis()-startTime);
+    info("access-log", sb.toString());
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java:44]: Instead of using string builder pattern, can we use simple log format? If we that way, in future it would be easy to extend and expose that and let users choose/change their format.

File: core/src/main/java/com/flipkart/gjex/core/filter/http/AccessLogHttpFilter.java [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+    this.startTime = startTime;
+  }
+
+  public StringBuilder getSb() {
+    return sb;
+  }
+
+  public void setSb(StringBuilder sb) {
+    this.sb = sb;
+  }
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/http/AccessLogHttpFilter.java:72]: Is this really required?

File: core/src/main/java/com/flipkart/gjex/core/filter/http/HttpFilter.java [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * An Http Filter extending {@link Filter}  interface for processing Request, Request-Headers,
+ * Response and
+ * Response-Headers
+ * around HTTP method invocation
+ *
+ * @author ajay.jalgaonkar
+ */
+
+@Data
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/http/HttpFilter.java:29]: Why does this need to have `Data` annotation? This is not a pojo class.

File: core/src/main/java/com/flipkart/gjex/core/filter/http/HttpFilterParams.java [hunk 27]
```diff
@@ -0,0 +1,17 @@
...
+ * Parameters for creating http filters
+ * @author ajay.jalgaonkar
+ *
+ */
+
+@Data
+public class HttpFilterParams {
+  private final Filter filter;
+  private final String pathSpec;
+}
```
Comment (kingster) [inline core/src/main/java/com/flipkart/gjex/core/filter/http/HttpFilterParams.java:17]: We already have `RequestParams` which has simlar, can we merge?

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGrpcFilter.java [hunk 28]
```diff
@@ -15,27 +15,25 @@
...
+ * An implementation of the {@link GrpcFilter} interface as example that performs naive authentication based on
  * information contained in the Request headers
  *
  * @author regu.b
  *
  */
-@Named("AuthFilter")
-public class AuthFilter implements Filter<HelloRequest, HelloReply>, Logging {
+@Named("AuthGrpcFilter")
+public class AuthGrpcFilter extends GrpcFilter<HelloRequest, HelloReply> {
```
Comment (kingster) [inline examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGrpcFilter.java:36]: This can remain `AuthFilter`

Lets not change the grpc filter names. Gjex is major for grpc and http is provided for migration period. In long term i don't see the need for having this explicit name in filter.s

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGrpcFilter.java [hunk 29]
```diff
@@ -15,31 +15,33 @@
...
 
 /**
- * An implementation of the {@link Filter} interface as example that simply logs Request information
+ * An implementation of the {@link GrpcFilter} interface as example that simply logs Request information
  * @author regu.b
  */
-@Named("LoggingFilter")
-public class LoggingFilter<Req extends GeneratedMessageV3, Res extends GeneratedMessageV3> implements Filter<Req, Res>, Logging {
+@Named("LoggingGrpcFilter")
+public class LoggingGrpcFilter<Req extends GeneratedMessageV3,
```
Comment (kingster) [inline examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGrpcFilter.java:31]: This can remain `LoggingFilter`

Lets not change the grpc filter names. Gjex is major for grpc and http is provided for migration period. In long term i don't see the need for having this explicit name in filter.s

**Review by kingster — CHANGES_REQUESTED** [review]

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGrpcFilter.java (hunk 29 above)
Comment (ajaypj) [reply examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/LoggingGrpcFilter.java:31]: done

File: examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGrpcFilter.java (hunk 28 above)
Comment (ajaypj) [reply examples/src/main/java/com/flipkart/gjex/examples/helloworld/filter/AuthGrpcFilter.java:36]: done

File: core/src/main/java/com/flipkart/gjex/core/filter/http/AccessLogHttpFilter.java (hunk 25 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/http/AccessLogHttpFilter.java:72]: removed

File: core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java (hunk 23 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/grpc/AccessLogGrpcFilter.java:74]: removed

File: core/src/main/java/com/flipkart/gjex/core/filter/Filter.java (hunk 21 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/filter/Filter.java:24]: reset

File: core/src/main/java/com/flipkart/gjex/core/config/GrpcConfig.java (hunk 20 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/config/GrpcConfig.java:35]: updated

File: core/src/main/java/com/flipkart/gjex/core/config/ApiService.java (hunk 19 above)
Comment (ajaypj) [reply core/src/main/java/com/flipkart/gjex/core/config/ApiService.java:44]: we should keep filterconfig as separate config for grpc and http. so it can't be top level.
in future other updates to filterconfig can come

**Review by kingster — APPROVED** [review]
